### ISSUE TITLE
If case expression

### ISSUE THREAD

**Issue by seebees** [dafny-lang/dafny#3131]
### What is the feature you would like to see in a future version of Dafny?

Currently in dafny you can use an `if case` as a statement

```dafny

if
  case x.Foo? => {

  }
  case x.Bar? => {

  }

```

I would like to use this syntax as an expression 

```dafny
if
  case x.Foo? then


  case x.Bar? then

 ```

My understanding is that there is some work around determinate behavior or something.

**Comment by cpitclaudel** [comment]
> My understanding is that there is some work around determinate behavior or something.

That's right: this can only be allowed in a compiled function if we can prove that the code is deterministic.
